clients/ts/signalr/src/HubConnectionBuilder.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { HttpConnection, IHttpConnectionOptions } from "./HttpConnection";
+import { HubConnection, JsonHubProtocol } from "./HubConnection";
+import { IHubProtocol } from "./IHubProtocol";
+import { ILogger, LogLevel } from "./ILogger";
+import { HttpTransportType } from "./ITransport";
+import { NullLogger } from "./Loggers";
+import { Arg, ConsoleLogger } from "./Utils";
+
+export class HubConnectionBuilder {
+    /** @internal */
+    public protocol: IHubProtocol;
+    /** @internal */
+    public httpConnectionOptions: IHttpConnectionOptions;
+    /** @internal */
+    public url: string;
+    /** @internal */
+    public logger: ILogger;
+
+    public configureLogging(logging: LogLevel | ILogger): HubConnectionBuilder {
+        Arg.isRequired(logging, "logging");
+
+        if (isLogger(logging)) {
+            this.logger = logging;
+        } else {
+            this.logger = new ConsoleLogger(logging);
+        }
+
+        return this;
+    }
+
+    public withUrl(url: string): HubConnectionBuilder;
+    public withUrl(url: string, options: IHttpConnectionOptions): HubConnectionBuilder;
+    public withUrl(url: string, transportType: HttpTransportType): HubConnectionBuilder;
+    public withUrl(url: string, transportTypeOrOptions?: IHttpConnectionOptions | HttpTransportType): HubConnectionBuilder {
```

Comment:
I can't find evidence pro or con this as a TypeScript idiom, but I like it. It's clearer for us when we're working on the code. The name `transportTypeOrOptions` does not appear in IntelliSense at all.

Reply:
https://blog.mariusschulz.com/2016/08/18/function-overloads-in-typescript provides some support, but is just a rando blog I found ;)